bindings/rust/bench/certs/generate_certs.sh

```diff
@@ -6,36 +6,62 @@
 # immediately bail if any command fails
 set -e
 
+# go to directory script is located in
 pushd "$(dirname "$0")"
 
-echo "generating CA private key and certificate"
-openssl req -nodes -new -x509 -keyout ca-key.pem -out ca-cert.pem -days 65536 -config config/ca.cnf
+# Generates certs with given algorithms and bits in $1$2/, ex. ec384/
+# $1: rsa or ec
+# $2: number of bits
+cert-gen () {
+    echo -e "\n----- generating certs for $1$2 -----\n"
 
-# secp384r1 is an arbitrarily chosen curve that is supported by the default
-# security policy in s2n-tls.
-# https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md#chart-security-policy-version-to-supported-curvesgroups
-echo "generating server private key and CSR"
-openssl req  -new -nodes -newkey ec -pkeyopt ec_paramgen_curve:secp384r1 -keyout server-key.pem -out server.csr -config config/server.cnf
+    # set openssl argument name
+    if [[ $1 == rsa ]]; then
+        local argname=rsa_keygen_bits:
+    elif [[ $1 == ec ]]; then
+        local argname=ec_paramgen_curve:P-
+    fi
 
-echo "generating client private key and CSR"
-openssl req  -new -nodes -newkey ec -pkeyopt ec_paramgen_curve:secp384r1 -keyout client-key.pem -out client.csr -config config/client.cnf
+    # make directory for certs
+    if [ ! -d $1$2/ ]; then
+        mkdir $1$2
+    fi
+    cd $1$2
 
-echo "generating server certificate and signing it"
-openssl x509 -days 65536 -req -in server.csr -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem -extensions req_ext -extfile config/server.cnf
+    echo "generating CA private key and certificate"
+    openssl req -new -nodes -x509 -newkey $1 -pkeyopt $argname$2 -keyout  ca-key.pem -out ca-cert.pem -days 65536 -config ../config/ca.cnf
 
-echo "generating client certificate and signing it"
-openssl x509 -days 65536 -req -in client.csr -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out client-cert.pem -extensions req_ext -extfile config/client.cnf
+    echo "generating server private key and CSR"
+    openssl req  -new -nodes -newkey $1 -pkeyopt $argname$2 -keyout server-key.pem -out server.csr -config ../config/server.cnf
 
-echo "verifying generated certificates"
-openssl verify -CAfile ca-cert.pem server-cert.pem
-openssl verify -CAfile ca-cert.pem client-cert.pem
+    echo "generating client private key and CSR"
```

Comment:
Could you reuse the same certs for server auth as for client auth?

Reply:
We could and it works, but it feels weird to. Maybe the crypto backend would cache the common verification things it needs to do for both the server and client for mTLS? It just doesn't mirror real use cases, and generating two sets of certs, one for the client and one for the server, isn't that much more work.